### PR TITLE
support grant statement propagation for aggregates

### PR DESCRIPTION
**Pull request by aykut-bozkurt**
DESCRIPTION: Adds support for GRANT/REVOKE ON aggregates.

support grant statement propagation for aggregates. This is a subtask for PR #6018.

### CODE REVIEW

File: src/backend/distributed/deparser/deparse_function_stmts.c [hunk 1]
```diff
@@ -789,7 +789,14 @@ static void
 AppendGrantOnFunctionFunctions(StringInfo buf, GrantStmt *stmt)
 {
 	ListCell *cell = NULL;
-	appendStringInfo(buf, " ON %s ", ObjectTypeToKeyword(stmt->objtype));
```
Comment (onderkalaci) [inline src/backend/distributed/deparser/deparse_function_stmts.c:792]: ObjectTypeToKeyword seems to be used in many other places.

1) Should we change ObjectTypeToKeyword instead?
2) Do we need anything to support those code-paths with aggragates?

File: src/backend/distributed/deparser/deparse_function_stmts.c [hunk 2]
```diff
@@ -789,7 +789,14 @@ static void
...
+
+	/*
+	 * The FUNCTION syntax works for plain functions, aggregate functions, and window
+	 * functions, but not for procedures; use PROCEDURE for those. Alternatively, use
+	 * ROUTINE to refer to a function, aggregate function, window function, or procedure
+	 * regardless of its precise type.
+	 */
```
Comment (onderkalaci) [inline src/backend/distributed/deparser/deparse_function_stmts.c:798]: definitely needs test. 

And, I'm not %100 sure if ROUTINE covers aggs for all cases. Where did you find that information? Would be good to link here as well

```SQL
alter routine taxi set owner to onderkalaci;
ERROR:  "public.taxi" is an aggregate function
```

File: src/backend/distributed/deparser/deparse_function_stmts.c (hunk 1 above)
Comment (aykut-bozkurt) [reply src/backend/distributed/deparser/deparse_function_stmts.c:792]: 1. objecttypetokeyword definition is correct. We do not need to use it for GRANT/REVOKE statements which can use [ROUTINE](https://www.postgresql.org/docs/current/sql-grant.html) keyword to support functions, procedures and aggregates.
2. Required changes are done to support GRANT/REVOKE for aggregates.

File: src/backend/distributed/deparser/deparse_function_stmts.c (hunk 2 above)
Comment (aykut-bozkurt) [reply src/backend/distributed/deparser/deparse_function_stmts.c:798]: The ROUTINE is only intended to be used in GRANT/REVOKE statements. I add tests.

**Review by onderkalaci — APPROVED** [review]
I guess it was simply overlooked that we have not support for `grant` on `aggregates`. So, it is also useful to fill the gap in GRANTs.

I think this PR deserves a specific PR DESCRIPTION so that users can see that we support this now

**Review by naisila — APPROVED** [review]

**Pull request merged**